### ISSUE TITLE
C# -> VB: Generics type parameter missed BC30737 and BC32042

### ISSUE THREAD

**Issue by jonr3** [icsharpcode/CodeConverter#682]
### Input code
```csharp
If relevant, please enter some example code here to reproduce the issue, or the steps taken to cause the issue.
```

### Erroneous output
[TaskCompletionSource2.zip](https://github.com/icsharpcode/CodeConverter/files/5579364/TaskCompletionSource2.zip)

```vbnet
            Dim tcs As TaskCompletionSource = Nothing

errors:

Severity	Code	Description	Project	File	Line	Suppression State
Error	BC32042	Too few type arguments to 'TaskCompletionSource(Of TResult)'.	TaskCompletionSource2	D:\clients\SADS\AsyncAwait\TaskCompletionSource2\TaskCompletionSource2\SdkProxy.vb	27	Active
```

### Expected output
```vbnet
I believe
Dim tcs As TaskCompletionSource(Of Boolean) = Nothing
```

### Details
* Product in use:  VS extension, VS2019 16.8.2
* Version in use: e.g. 8.2.1.0
* Did you see it working in a previous version, which? No, not used previous versions.
* Any other relevant information to the issue, or your interest in contributing a fix.

the Main sub is not correctly configured in the project so I also get this error:

Severity	Code	Description	Project	File	Line	Suppression State
Error	BC30737	No accessible 'Main' method with an appropriate signature was found in 'TaskCompletionSource2'.	TaskCompletionSource2	D:\clients\SADS\AsyncAwait\TaskCompletionSource2\TaskCompletionSource2\vbc	1	Active

whereas the c# code works fine with no changes.

**Comment by GrahamTheCoder** [comment]
Thanks for the report. Both issues should be straightforward to fix, it's just a case of finding time. I'll prioritise the generics one since it can cause lots of work on a big project whereas there's only one main method

**Comment by jonr3** [comment]
Thanks Graham,

The fix for the missing task type is:

Dim tcs As TaskCompletionSource(Of Boolean) = Nothing

Which compiles, as to whether its correct to allocate to nothing is another matter. The class TaskCompletionSource appears to require one and only one argument.

The duplicate “main” procedures are fixed by changing the private shared call to:

Public Shared Sub main(ByVal args As String())

It doesn’t seem to matter that the project points to the correct “main” as regardless of whether I choose “Sub Main” or TaskCompletionSource2.program the code works as it does in C# to the end.

I have a workaround, there is no hurry for a fix for this. I just wanted to let someone know

I have spent years using Telerik’s code converter … this extension is way better.

In earlier VS (2017) the code would be placed alongside the existing, in 2019 it replaces the whole rewritten project. I am not too keen on the idea of a project replacing the source, I wonder whether a folder could be created to contain the whole converted project to sit alongside the working project but be named  TaskCompletionSource2.vb  or TaskCompletionSource2.cs respectively for conversion to vb or c# or even as a pair of subfolders of the existing perhaps. Just my two pennyworth idea ..

Many thanks,

Regards

jON

**Comment by jonr3** [comment]
In fact a quick check confirms that the “nothing” causes the program to crash …

What is needed is:

Dim tcs As New TaskCompletionSource(Of Boolean)

Instead

Regards

jON

**Comment by GrahamTheCoder** [comment]
In terms of creating the new files alongside, that should still be what's happening: in general there should be no difference between 2017 and 2019. The solution file and any referencing projects are overwritten, but there are backups of the originals written by default.
I'd highly recommend storing your code in source control though which will make it trivial to pick and choose which changes you want to keep/discard. A few more notes on that here: https://github.com/icsharpcode/CodeConverter/wiki

**Comment by GrahamTheCoder** [comment]
Looks like this was fixed in #692
